rust/worker/src/execution/orchestration/common.rs

```diff
@@ -31,7 +31,12 @@ pub(super) async fn get_hnsw_segment_by_id(
     collection_id: &CollectionUuid,
 ) -> Result<Segment, Box<GetHnswSegmentByIdError>> {
     let segments = sysdb
-        .get_segments(Some(*hnsw_segment_id), None, None, *collection_id)
+        .get_segments(
+            Some(SegmentUuid(*hnsw_segment_id)),
+            None,
+            None,
+            *collection_id,
+        )
         .await;
```

Comment:
Ideally `SegmentUuid` should always be used as the identifier of segment. It should be parsed right after the request enters the worker and serialized right before it is sent as part of a request.

This implies we should not use an untyped `Uuid` as the argument of the method here, but since we are deprecating this soon it's fine to leave it here for now.